src/stores/habits.store.ts

```
@@ -21,57 +22,59 @@ type HabitsState = {
-      fetchHabits: async () => {
-        const habits = await listHabits();
-        set({ habits });
-      },
+        fetchHabits: async () => {
+          const habits = await listHabits();
 
-      removeHabit: async ({ iconPath, id }: Habit) => {
-        await destroyHabit(id);
+          const habitsMap = habits.reduce((acc, habit) => {
+            return {
+              ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)